src/stdatamodels/jwst/datamodels/tests/test_read_metadata.py

```diff
@@ -0,0 +1,91 @@
+"""Test utilities for loading metadata without loading entire model."""
+
+import pytest
+import numpy as np
+from sys import getsizeof
+from astropy.time import Time
+import stdatamodels.jwst.datamodels as dm
+from stdatamodels.jwst.datamodels.util import read_metadata
+
+
+TESTFILE_ROOT = "jwst_image."
+FILT = "GR150R"
+INPUT_TIME = "2021-01-01 00:00:00.000"
+DATA = np.zeros((1000, 1000), dtype=np.float32)
+
+
+@pytest.fixture
+def make_and_save_models(tmp_path):
+    """Create a new datamodel with a meta attribute and relatively large data array."""
+    model = dm.ImageModel(DATA)
+    model.meta.instrument.filter = FILT
+
+    # give this an astropy time to ensure it is cast to string
+    model.meta.observation.date = Time(INPUT_TIME, format="iso", scale="utc")
+
+    for extension in ["fits", "asdf"]:
+        # Save the datamodel to a file
+        root = TESTFILE_ROOT + extension
+        path = str(tmp_path / root)
+        model.save(path)
+        model.close()
+
+
+@pytest.mark.parametrize("extension", ["fits", "asdf"])
+def test_read_metadata(make_and_save_models, extension, tmp_path):
+    """Test metadata loading."""
+    root = TESTFILE_ROOT + extension
+    path = str(tmp_path / root)
+
+    # Load the metadata
+    meta = read_metadata(path)
+
+    # Ensure the dict is flat, i.e., none of the vals in keys are themselves dicts
+    assert isinstance(meta, dict)
+    assert all(not isinstance(v, dict) for v in meta.values())
+
+    # Ensure the metadata has the expected attributes
+    # with times re-cast from astropy Time to string (and not asdf.tagged.TaggedString)
+    assert isinstance(meta["meta.instrument.filter"], str)
+    assert isinstance(meta["meta.observation.date"], str)
+    assert meta["meta.instrument.filter"] == FILT
+    assert meta["meta.observation.date"] == INPUT_TIME
+    assert meta["meta.filename"] == root
+
+    # Ensure attributes in schema but not in model are there but give None
+    assert meta["meta.instrument.detector"] is None
+
+    # Ensure the metadata does not have data attributes
+    assert "data" not in meta
+    assert getsizeof(meta) < getsizeof(DATA)
+
+    # test flatten=False mode
```

Comment:
```suggestion
    # test flatten=False mode
```
Can we split this up into multiple tests. We don't need to go as far as 1 assert per test but fewer would be easier to understand what each assert is testing (based on the test name) and make refactoring easier.